xrmocap/transform/keypoints3d/optim/rm_duplicate.py

```diff
@@ -0,0 +1,143 @@
+# yapf: disable
+import logging
+import numpy as np
+import torch
+from typing import Union
+
+from xrmocap.data_structure.keypoints import Keypoints
+from xrmocap.ops.top_down_association.identity_tracking.builder import (
+    BaseTracking, build_identity_tracking,
+)
+from .base_optimizer import BaseOptimizer
+
+# yapf: enable
+
+
+class RemoveDuplicate(BaseOptimizer):
+    """This optimization tool helps remove duplicate identities by the L2
+    distance between 3D keypoints for each person and add tracked identities to
+    optimized 3D keypoints."""
+
+    def __init__(
+        self,
+        verbose: bool = True,
+        threshold: float = 2.0,
+        keep: str = 'by_index',
+        identity_tracking: Union[None, dict, BaseTracking] = None,
+        logger: Union[None, str, logging.Logger] = None,
+    ):
+        """Initialization for RemoveDuplicate optimizor.
+
+        Args:
+            verbose (bool, optional):
+                Whether to log info. Defaults to True.
+            threshold (float, optional):
+                Threshold for duplicate 3D keypoints. Defaults to 2.0.
+            keep (str, optional):
+                Mode to keep 3D keypoints, 'by_index' keeps the lowest index,
+                'by_conf' keeps the highest confidence. Defaults to 'by_index'.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+        """
+
+        super().__init__(verbose=verbose, logger=logger)
+        self.threshold = threshold
+        self.keep = keep
```

Comment:
`assert self.keep in {'by_index', 'by_conf'}`?
Otherwise L96 may raise reference before assignment error

Reply:
Fixed in [c2ba151](https://github.com/openxrlab/xrmocap/pull/46/commits/c2ba1517e1c5c68dd276bf21e002f6e4ef4df715).